lib/nconf/common.js

```diff
@@ -164,8 +164,12 @@ common.transform = function(map, fn) {
   });
 
 
-  return pairs.reduce(function(accumulator, pair) {
-    accumulator[pair.key] = pair.value;
```

Comment:
If `pair` equals `null`, trying to access its `key` property will throw a TypeError.